test/api.test.js

```diff
@@ -535,6 +545,74 @@ describe("api.js", () => {
             }]);
         });
 
+        it("should fail by timeout during rendition processing", async () => {
+            delete process.env.DISABLE_IO_EVENTS_ON_TIMEOUT;
+
+            const receivedMetrics = MetricsTestHelper.mockNewRelic();
+            testUtil.nockIOEvent({
+                type: "rendition_failed",
+                errorReason: "GenericError",
+                errorMessage: 'Timeout',
+                rendition: {
+                    fmt: "png"
+                },
+                source: "https://example.com/MySourceFile.jpg"
+            });
+
+            const main = worker(async function(source, rendition) {
+                fs.writeFileSync(rendition.path, testUtil.RENDITION_CONTENT);
+                await sleep(500);
+                console.log('waiting...');
+                return Promise.resolve();
+            });
+            assert.equal(typeof main, "function");
+            process.env.__OW_DEADLINE = Date.now() + 300;
+
+            await main(testUtil.simpleParams({noEventsNock:true}));
+            testUtil.assertNockDone();
+            await MetricsTestHelper.metricsDone();
+            MetricsTestHelper.assertArrayContains(receivedMetrics, [{
+                eventType: 'timeout'
+            },{
+                eventType: "activation"
+            }]);
+            assert.equal(processSpy.calledOnce, true, "did not call process.exit(124) on timeout");
+        });
+
+        it("should fail by timeout during second rendition processing", async () => {
+            delete process.env.DISABLE_IO_EVENTS_ON_TIMEOUT;
+            const receivedMetrics = MetricsTestHelper.mockNewRelic();
+            testUtil.nockIOEvent({
+                type: "rendition_failed",
+                errorReason: "GenericError",
+                errorMessage: 'Timeout',
+                rendition: { fmt: 'xml', name: 'MyRendition3.xml' },
+                source: "https://example.com/MySourceFile.jpg"
+            });
+
+            const main = worker(async function(source, rendition) {
+                if (rendition.index === 2) {
+                    console.log('waiting...');
+                    await sleep(5000);
+                }
+                fs.writeFileSync(rendition.path, testUtil.RENDITION_CONTENT);
+                return Promise.resolve();
+            });
+            assert.equal(typeof main, "function");
+            process.env.__OW_DEADLINE = Date.now() + 7000;
```

Comment:
Same here, 7 seconds is very long, why not 100ms (and adapting the other sleep value accordingly)?

Reply:
The reason I chose 7s here is that I am testing a timeout in the middle of processing. The goal of this test is to process 2 renditions successfully and the third one times out. The test should send 2 rendition created events and 1 rendition failed event. (The reason i used 3 renditions instead of 2 is that the [multiple rendition params function](https://github.com/adobe/asset-compute-sdk/blob/a0df461eedd7fb55d760d04b19f3ab0e12cd20ae/test/testutil.js#L187) from the test utils had 3 renditions). To successfully do this, I need the timeout to be > 5s since that is the [timeout buffer](https://github.com/adobe/asset-compute-sdk/blob/a0df461eedd7fb55d760d04b19f3ab0e12cd20ae/lib/worker.js#L35). If the action deadline is in <5s, the action will start its timeout process during the first rendition.

One thing I can try is using sinon fake timers to skip ahead in time instead of sleeping for 5s so the test will not actually take 5+ seconds to run

Reply:
> I need the timeout to be > 5s since that is the timeout buffer. If the action deadline is in <5s, the action will start its timeout process during the first rendition.

I see. I am ok with that then (if fake timers don't work out).